src/dachshund/candidate.rs

```diff
@@ -403,6 +418,45 @@ impl<'a, TGraph: GraphBase> Candidate<'a, TGraph> {
         Ok(cliqueness)
     }
 
+    // Returns true if every core node has at least thresh fraction
+    // of the possible edges, using/updating the local density guarantee
+    // as applicable.
+    pub fn local_thresh_score_at_least(&mut self, thresh: f32) -> bool {
+        if thresh == 0.0 {
+            return true
+        }
+
+        let previous_thresh : f32;
+        let nodes_to_check;
+
+        match &self.local_guarantee {
+            None => {
+                previous_thresh = 1.0;
+                nodes_to_check = &self.core_ids;
+            },
+            Some(guarantee) => {
+                previous_thresh = guarantee.thresh;
+                nodes_to_check = if previous_thresh >= thresh
+                    {&guarantee.exceptions} else {&self.core_ids};
+            },
+        };
+
+        for &node_id in nodes_to_check {
+            // [TODO] This can be refactored to return the actual number
+            // to allow us to store a tighter guarantee.
+            if !self.get_node(node_id).get_local_thresh_score(
+                thresh,
+                &self.non_core_ids,
+                self.max_core_node_edges,
+            ){ return false }
+        }
+        self.local_guarantee = Some(LocalDensityGuarantee{
+                thresh: previous_thresh.min(thresh),
```

Comment:
is this always optimal? If previous_thresh was higher but with few nodes in exceptions, could we end up with a quicker runtime by keeping that higher thresh?

Reply:
Yes, in principle this update could result in a correct but less useful guarantee. In practice, we're always testing with the same threshold, so I think this will always be optimal.